shell/platform/embedder/BUILD.gn

```diff
@@ -14,49 +14,68 @@ shell_gpu_configuration("embedder_gpu_configuration") {
   enable_metal = false
 }
 
-source_set("embedder") {
-  sources = [
-    "embedder.cc",
-    "embedder.h",
-    "embedder_engine.cc",
-    "embedder_engine.h",
-    "embedder_external_texture_gl.cc",
-    "embedder_external_texture_gl.h",
-    "embedder_include.c",
-    "embedder_platform_message_response.cc",
-    "embedder_platform_message_response.h",
-    "embedder_safe_access.h",
-    "embedder_surface.cc",
-    "embedder_surface.h",
-    "embedder_surface_gl.cc",
-    "embedder_surface_gl.h",
-    "embedder_surface_software.cc",
-    "embedder_surface_software.h",
-    "embedder_task_runner.cc",
-    "embedder_task_runner.h",
-    "embedder_thread_host.cc",
-    "embedder_thread_host.h",
-    "platform_view_embedder.cc",
-    "platform_view_embedder.h",
-    "vsync_waiter_embedder.cc",
-    "vsync_waiter_embedder.h",
-  ]
+# Template for the embedder build. Used to allow building it multiple times with
+# different flags.
+template("embedder_source_set") {
```

Comment:
It may be better to create a template for the embedder dylib (i.e a template that expands to `shared_library` instead of `source_set`) that not only sets the macro to rename symbols but also adds a suffix to the dylib so that multiple dylib versions with renamed symbols may coexist in the out directory.

Reply:
What's the use case for this? I don't expect us to ever need to build a prefixed dylib.